File: src/07.greedy/01.arrayGreedy/11.56.ts
export function merge(intervals: number[][]): number[][] {
    // 排序
    intervals.sort((a, b) => a[0] - b[0]);

    let i = 0;

    for (let j = 1; j < intervals.length; j++) {
        if (intervals[i][1] >= intervals[j][0]) {
            intervals[i][1] = Math.max(intervals[i][1], intervals[j][1]);
        } else {
            i++;
            intervals[i] = intervals[j];
        }
    }

    return intervals.slice(0, i + 1);
}

File: src/07.greedy/01.arrayGreedy/11.56.go
package array_greedy

import "sort"

func Merge(intervals [][]int) [][]int {

	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})

	idx := 0
	for i := 1; i < len(intervals); i++ {
		if intervals[idx][1] >= intervals[i][0] {
			if intervals[idx][1] < intervals[i][1] {
				intervals[idx][1] = intervals[i][1]
			}
		} else {
			idx++
			intervals[idx] = intervals[i]
		}
	}

	return intervals[:idx+1]
}
